File: app/src/main/java/fr/corenting/edcompanion/fragments/GalnetFragment.java
package fr.corenting.edcompanion.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;

import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

import fr.corenting.edcompanion.R;
import fr.corenting.edcompanion.adapters.NewsAdapter;
import fr.corenting.edcompanion.models.events.GalnetNews;
import fr.corenting.edcompanion.network.NewsNetwork;
import fr.corenting.edcompanion.utils.NotificationsUtils;
import fr.corenting.edcompanion.utils.SettingsUtils;

public class GalnetFragment extends AbstractListFragment<NewsAdapter> {

    public static final String GALNET_FRAGMENT_TAG = "galnet_fragment";

    private String language;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        language = getNewsLanguage();
    }

    private String getNewsLanguage() {
        return SettingsUtils.getString(getContext(),
                getString(R.string.settings_galnet_lang));
    }

    @Override
    public void onResume() {
        super.onResume();

        // Refresh if language changed
        if (!language.equals(getNewsLanguage())) {
            language = getNewsLanguage();
            getData();
        }
    }

    @Override
    void getData() {
        NewsNetwork.getGalnetNews(getContext(), language);
    }

    @Override
    NewsAdapter getAdapter() {
        return new NewsAdapter(getContext(), recyclerView, false);
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onNewsEvent(GalnetNews news) {
        // Error case
        if (!news.getSuccess()) {
            endLoading(true);
            NotificationsUtils.displayGenericDownloadErrorSnackbar(getActivity());
            return;
        }

        endLoading(news.getArticles().size() == 0);
        recyclerViewAdapter.submitList(news.getArticles());
    }
}

File: app/src/main/java/com/github/masdaster/edma/fragments/GalnetFragment.kt
package com.github.masdaster.edma.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.activityViewModels
import com.github.masdaster.edma.R
import com.github.masdaster.edma.adapters.NewsAdapter
import com.github.masdaster.edma.models.ProxyResult
import com.github.masdaster.edma.models.events.News
import com.github.masdaster.edma.utils.NotificationsUtils
import com.github.masdaster.edma.utils.SettingsUtils
import com.github.masdaster.edma.view_models.GalnetViewModel

class GalnetFragment : AbstractListFragment<NewsAdapter>() {
    private lateinit var currentLanguage: String
    private val viewModel: GalnetViewModel by activityViewModels()

    private val newsLanguage: String
        get() = SettingsUtils.getString(context, getString(R.string.settings_news_lang))

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        currentLanguage = newsLanguage
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = super.onCreateView(inflater, container, savedInstanceState)

        viewModel.getGalnet().observe(viewLifecycleOwner, ::onNewsEvent)
        return view
    }


    override fun onResume() {
        super.onResume()

        // Refresh if language changed
        if (currentLanguage != newsLanguage) {
            getData()
        }
    }

    override fun getData() {
        viewModel.fetchGalnet(currentLanguage)
    }

    override fun getNewRecyclerViewAdapter(): NewsAdapter {
        return NewsAdapter(context, binding.recyclerView, false, true)
    }

    private fun onNewsEvent(news: ProxyResult<News>) {
        // Error case
        if (news.error != null || news.data == null) {
            endLoading(true)
            NotificationsUtils.displayGenericDownloadErrorSnackbar(activity)
            return
        }

        endLoading(news.data.articles.isEmpty())
        recyclerViewAdapter.submitList(news.data.articles)
    }

    companion object {
        const val GALNET_FRAGMENT_TAG = "galnet_fragment"
    }
}
